docs/book/src/cronjob-tutorial/testdata/project/cmd/main.go

```diff
@@ -135,7 +137,12 @@ func main() {
 			// These configurations ensure that only authorized users and service accounts
 			// can access the metrics endpoint. The RBAC are configured in 'config/rbac/kustomization.yaml'. More info:
 			// https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.18.4/pkg/metrics/filters#WithAuthenticationAndAuthorization
-			FilterProvider: filters.WithAuthenticationAndAuthorization,
+			FilterProvider: func() func(c *rest.Config, httpClient *http.Client) (metricsserver.Filter, error) {
+				if secureMetrics {
+					return filters.WithAuthenticationAndAuthorization
+				}
+				return nil
+			}(),
```

Comment:
Based on the issue, it seems that we reach out the consensus to keep it simple (KISS) with something like

```
if secureMetrics {
         FilterProvider: func() metricsserver.FilterProvider {
            return filters.WithAuthenticationAndAuthorization
        }   
    },
```

Did you try to do in this way?

Reply:
There's no such type in that module.